org.eclipse.jdt.core/s4jie2-tests/runner/org/eclipse/jdt/core/s4jie2TestSuite.java

```diff
@@ -233,9 +283,203 @@ public static void main(String[] args) throws IOException {
+				+ "	at Foo.bar(correct_may_throw_exception.java:7)\n"
+				+ "	at Main.main(correct_may_throw_exception.java:14)\n");
+		testCompileAndRun(true, "correct_throw_may_throw_exception", false, "",
+				"Exception in thread \"main\" java.lang.IllegalArgumentException\n"
+				+ "	at Foo.bar(correct_throw_may_throw_exception.java:8)\n"
+				+ "	at Foo.bar$spec(correct_throw_may_throw_exception.java)\n"
+				+ "	at Main.main(correct_throw_may_throw_exception.java:15)\n");
+		testCompileAndRun(true, "no_exception_throw_condition_satisfied", false, 
+				"",
+				"Exception in thread \"main\" java.lang.AssertionError: @throws condition holds but specified exception type not thrown\n"
+				+ "	at Main.foo$post(no_exception_throw_condition_satisfied.java:6)\n"
+				+ "	at Main.foo(no_exception_throw_condition_satisfied.java:7)\n"
+				+ "	at Main.main(no_exception_throw_condition_satisfied.java:11)\n");
+		testPartOfStringCompileAndRun(true, "wrong_exception_throw_condition_satisfied", false, "", 
+				"SEVERE: @throws condition holds but specified exception type not thrown\n");
+		testPartOfStringCompileAndRun(true, "wrong_exception_throw_condition_satisfied", false, "", 
+				"SEVERE: The thrown exception was not specified in the formal specification\n"
+				+ "Exception in thread \"main\" java.lang.ArithmeticException\n"
+				+ "	at Main.foo(wrong_exception_throw_condition_satisfied.java:7)\n"
+				+ "	at Main.main(wrong_exception_throw_condition_satisfied.java:11)\n");
+		testCompileAndRun(true, "no_throw_may_throw_condition_satisfied", true, "", "");
+		testCompileAndRun(true, "no_throw_may_throw_condition_not_satisfied", true, "", "");
+		testCompileAndRun(true, "multiple_throw_conditions_satisfied_no_throw", false, "",
+				"Exception in thread \"main\" java.lang.AssertionError: @throws condition holds but specified exception type not thrown\n"
+				+ "	at Main.foo$post(multiple_throw_conditions_satisfied_no_throw.java:7)\n"
+				+ "	at Main.foo(multiple_throw_conditions_satisfied_no_throw.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_satisfied_no_throw.java:12)\n");
+		testPartOfStringCompileAndRun(true, "multiple_throw_conditions_satisfied_throw_second", false, "",
+				"SEVERE: @throws condition holds but specified exception type not thrown\n"
+				+ "Exception in thread \"main\" java.lang.ArithmeticException\n"
+				+ "	at Main.foo(multiple_throw_conditions_satisfied_throw_second.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_satisfied_throw_second.java:12)\n");
+		testPartOfStringCompileAndRun(true, "multiple_throw_conditions_satisfied_throw_first", false, "",
+				"SEVERE: @throws condition holds but specified exception type not thrown\n"
+				+ "Exception in thread \"main\" java.lang.IllegalArgumentException\n"
+				+ "	at Main.foo(multiple_throw_conditions_satisfied_throw_first.java:8)\n"
+				+ "	at Main.main(multiple_throw_conditions_satisfied_throw_first.java:12)\n");
```

Comment:
This program should not generate any kind of error message; it complies with its specification. A sequence of `@throws` clauses should be read as "if C1, then throw E1, *else* if C2, then throw E2, *else* if C3, etc.". (Otherwise there would be no program that satisfies this spec?)